evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -0,0 +1,175 @@
+
+        nan_ratio = X.ww.describe().loc["nan_count"] / X.shape[0]
+        self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
+
+        def _filter_cols(impute_strat, X):
+            """Function to return which columns of the dataset to impute given the impute strategy."""
+            cols = []
+            if self.parameters["categorical_impute_strategy"] == impute_strat:
+                if self.parameters["numeric_impute_strategy"] == impute_strat:
+                    cols = list(X.columns)
+                else:
+                    cols = list(X.ww.select(exclude=["numeric"]).columns)
+            elif self.parameters["numeric_impute_strategy"] == impute_strat:
+                cols = list(X.ww.select(include=["numeric"]).columns)
+
+            X_cols = [col for col in cols if col not in self._all_null_cols]
+            if len(X_cols) > 0:
+                return X_cols
+
+        self._forwards_cols = _filter_cols("forwards_fill", X)
+        self._backwards_cols = _filter_cols("backwards_fill", X)
+        self._interpolate_cols = _filter_cols("interpolate", X)
+
+        if y is not None:
+            y = infer_feature_types(y)
+            if y.isnull().any():
+                self._impute_target = self.parameters["target_impute_strategy"]
+
+        return self
+
+    def transform(self, X, y=None):
+        """Transforms data X by imputing missing values using specified timeseries-specific strategies. 'None' values are converted to np.nan before imputation and are treated as the same.
+
+        Args:
+            X (pd.DataFrame): Data to transform
+            y (pd.Series, optional): Ignored.
+
+        Returns:
+            pd.DataFrame: Transformed X
+        """
+        X = infer_feature_types(X)
+        if len(self._all_null_cols) == X.shape[1]:
+            df = pd.DataFrame(index=X.index)
+            df.ww.init()
+            return df, y
+
+        X_no_all_null = X.ww.drop(self._all_null_cols)
+
+        if self._forwards_cols is not None:
+            X_forward = X.ww[self._forwards_cols]
+            imputed = X_forward.pad()
+            imputed.bfill(inplace=True)  # Fill in the first value, if missing
+            X_no_all_null[X_forward.columns] = imputed
+
+        if self._backwards_cols is not None:
+            X_backward = X.ww[self._backwards_cols]
+            imputed = X_backward.bfill()
+            imputed.pad(inplace=True)  # Fill in the last value, if missing
+            X_no_all_null[X_backward.columns] = imputed
+
+        if self._interpolate_cols is not None:
+            X_interpolate = X.ww[self._interpolate_cols]
+            imputed = X_interpolate.interpolate()
+            imputed.bfill(inplace=True)  # Fill in the first value, if missing
+            X_no_all_null[X_interpolate.columns] = imputed
+
+        y_imputed = pd.Series(y)
+        if self._impute_target == "forwards_fill":
+            y_imputed = y.pad()
+            y_imputed = y_imputed.bfill()
```

Comment:
For my own learning, is there a reason you do the X data imputation in place vs. the target?

Reply:
Nope, total oversight on my part. I'll update this to be inplace as well.